src/components/views/read/cvForm/Education.tsx

```diff
@@ -196,77 +202,44 @@ const Education: FC<Props> = ({ nextStep, prevStep }) => {
       ...prevErrors,
       [`${cardId}-${key}`]: '',
     }));
+    onError(false);
   };
+
   const handlePrev = (): void => {
     prevStep();
   };
   const handleNext = (): void => {
-    let isValid = true;
-    const updatedErrors = { ...errors };
-
     // Check each card for unfilled required fields
     educationCards?.forEach((card) => {
-      if (!card.data.degree.trim()) {
-        updatedErrors[`${card.id}-degree`] = 'Degree is required';
-        isValid = false;
-      } else {
-        updatedErrors[`${card.id}-degree`] = '';
-      }
-
-      if (!card.data.institutionName.trim()) {
-        updatedErrors[`${card.id}-institutionName`] =
-          'Institution Name is required';
-        isValid = false;
-      } else {
-        updatedErrors[`${card.id}-institutionName`] = '';
-      }
-
-      if (!card.data.major.trim()) {
-        updatedErrors[`${card.id}-major`] = 'Major is required';
-        isValid = false;
-      } else {
-        updatedErrors[`${card.id}-major`] = '';
-      }
-
-      if (!card.data.startDate?.trim()) {
-        updatedErrors[`${card.id}-startDate`] = 'Start Date is required';
-        isValid = false;
-      } else {
-        updatedErrors[`${card.id}-startDate`] = '';
-      }
-      if (!card.data.endDate?.trim()) {
-        updatedErrors[`${card.id}-endDate`] = 'End Date is required';
-        isValid = false;
-      } else {
-        updatedErrors[`${card.id}-endDate`] = '';
-      }
-
-      if (!card.data.country.trim()) {
-        updatedErrors[`${card.id}-country`] = 'Country is required';
-        isValid = false;
-      } else {
-        updatedErrors[`${card.id}-country`] = '';
-      }
-
-      // Check if the card has any unfilled required fields
-      if (!isValid) {
+      const result = suite(card.data);
+      if (result.hasErrors()) {
+        setIsValid(false);
+        // Handle validation errors
+        const updatedErrors = { ...errors };
+        Object.keys(result.tests).forEach((fieldName) => {
+          const fieldErrors = result.tests[fieldName].errors || [];
+          if (fieldErrors.length > 0) {
+            updatedErrors[`${card.id}-${fieldName}`] = fieldErrors[0] || '';
+          }
+        });
+        setErrors(updatedErrors);
         setShowFields((prevShowFields) => ({
           ...prevShowFields,
           [card.id]: true,
         }));
       }
     });
 
-    setErrors(updatedErrors);
-
     const allSaved = educationCards?.every((card) => card.data.saved);
 
     if (isValid && allSaved) {
       nextStep();
-    } else if (isValid && !allSaved) {
+    } else if (!isValid && !allSaved) {
```

Comment:
Do we need to modify the condition?

Reply:
Yes. this will not work as you expect.
Imagine at the beginning of your funciton that `isValid = false` if the state is updated with `setIsValid(true)` when you arrive at this condition you will still get `isValid = false`.
The state value will be `isValid = true` only at the next render. since you are inside the same render, it will not have been updated. You can however use a local variable. In this case question your use of the isValid state. If you do not use it for anything else, you might remove it.

Reply:
having isValid for this use, which is supposed to be a main point of checking whether the user is good to go and show that it's saved and all the fields are valid, then they can proceed on to next section or whatever.